### PR TITLE
Relax tolerances and avoid non-differentiable points for FP16 in triplet loss tests

### PR DESCRIPTION
**Pull request by hvy**

### CODE REVIEW

**Comment by crcrpar** [comment]
Does this relate to https://github.com/chainer/chainer/pull/5819 and #6190?

**Comment by hvy** [comment]
Ah, I did not know that it was addressed already. I think it's okay to solve it using the non-differentiable-dodging also.

**Comment by crcrpar** [comment]
If the pair of relaxation and dodge_nondifferentiable enough for F.triplet test to settle, then I would like 5819 to adopt this relaxation and get merged.

**Comment by hvy** [comment]
Yes, that might be better as the non-differentiable points are known.

File: tests/chainer_tests/functions_tests/loss_tests/test_triplet.py [hunk 1]
```diff
@@ -21,10 +21,13 @@
 class TestTriplet(unittest.TestCase):
 
     def setUp(self):
+        # Sample differentiable inputs
         eps = 1e-3
+        if self.dtype == numpy.float16:
+            eps *= 10
```
Comment (niboshi) [inline tests/chainer_tests/functions_tests/loss_tests/test_triplet.py]: Is there any reason you wrote this way instead of `eps = 1e-2`?

**Comment by niboshi** [comment]
Let's merge this PR first, as it's not clear if `dodge_nondifferentiable` works with this fix.

File: tests/chainer_tests/functions_tests/loss_tests/test_triplet.py (hunk 1 above)
Comment (hvy) [reply tests/chainer_tests/functions_tests/loss_tests/test_triplet.py]: No, either should be fine. Do you want me to change it?

Comment (niboshi) [reply tests/chainer_tests/functions_tests/loss_tests/test_triplet.py]: Yes, please.

Comment (hvy) [reply tests/chainer_tests/functions_tests/loss_tests/test_triplet.py]: Fixed.

**Comment by hvy** [comment]
PTAL.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM

**Pull request merged**